plugins/module_utils/network/sonic/facts/static_routes/static_routes.py

```diff
@@ -0,0 +1,173 @@
+        """ Populate the facts for static_routes
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+        objs = []
+        if connection:  # just for linting purposes, remove
+            pass
+
+        if not data:
+            static_routes_config = self.get_static_routes(self._module)
+            data = self.update_static_routes(static_routes_config)
+        # operate on a collection of resource x
+        for conf in data:
+            if conf:
+                obj = self.render_config(self.generated_spec, conf)
+        # split the config into instances of the resource
+                if obj:
+                    objs.append(obj)
+
+        ansible_facts['ansible_network_resources'].pop('static_routes', None)
+        facts = {}
+        if objs:
+            params = utils.validate_config(self.argument_spec, {'config': remove_empties_from_list(objs)})
+            facts['static_routes'] = params['config']
+        ansible_facts['ansible_network_resources'].update(facts)
+        return ansible_facts
+
+    def render_config(self, spec, conf):
+        """
+        Render config as dictionary structure and delete keys
+          from spec for null values
+        :param spec: The facts tree, generated from the argspec
+        :param conf: The configuration
+        :rtype: dictionary
+        :returns: The generated config
+        """
+
+        return conf
+
+    def get_static_routes(self, module):
+        all_static_routes = []
+        vrfs = get_all_vrfs(module)
+        for vrf_name in vrfs:
+            get_path = '%s=%s/%s' % (network_instance_path, vrf_name, protocol_static_routes_path)
+            request = {'path': get_path, 'method': 'get'}
+            try:
+                response = edit_config(module, to_request(module, request))
+            except ConnectionError as exc:
+                module.fail_json(msg=str(exc), code=exc.code)
+            for resp in response:
+                if 'openconfig-network-instance:static-routes' in resp[1]:
+                    static_routes_dict = resp[1].get('openconfig-network-instance:static-routes', {})
+                    static_routes_dict['vrf'] = vrf_name
+                    all_static_routes.append(static_routes_dict)
+        return all_static_routes
+
+    def update_static_routes(self, data):
+        static_vrf_list = []
+        for static_route in data:
+            static_vrf_dict = {}
+            static_route_list = static_route.get('static', [])
+            vrf_name = static_route.get('vrf', None)
+            static_list = []
+            for static in static_route_list:
+                static_dict = {}
+                prefix = static.get('prefix', None)
+                next_hops = static.get('next-hops', None)
+                next_hop_dict_list = next_hops.get('next-hop', [])
```

Comment:
Thanks for making the name change to distinguish the two lists of next hops (input list and parsed output dict list), but the usage here is confusing. (It looks like it is reversed from the intended usage.) Please reverse the usage for these two list names ("next_hop_dict_list" and "next_hops_list") to match the naming for the contained items. 

The payload list to which a parsed "next_hop_dict" should be appended is the "next_hop_dict_list".  For the list of next hops obtained from the input playbook, the previously used name "next_hops_list" was fine and should be restored.